spec/controllers/queued_posts_controller_spec.rb

```diff
@@ -31,6 +31,21 @@
     let!(:user) { log_in(:moderator) }
     let(:qp) { Fabricate(:queued_post) }
 
+    context 'not found' do
+      it 'returns json error' do
+        id = qp.id
+        qp.destroy
+
+        put :update, params: {
+          id: id, queued_post: { state: 'approved' }
+        }, format: :json
+
+        expect(response).not_to be_success
```

Comment:
I'm personally not a fan of using `not_to be_success` as it just means `response.status != 200`. Instead I would test directly for the right response status. Is it a 404, 403 or 500?